db/compaction.h

```diff
@@ -258,6 +258,8 @@ class Compaction {
   static bool IsFullCompaction(VersionStorageInfo* vstorage,
                                const std::vector<CompactionInputFiles>& inputs);
 
+  VersionStorageInfo* vstorage_;
```

Comment:
Maybe call it `input_vstorage_`.

Replace all the input_version_->storage_info() with this one.